framework/authentication.py

```diff
@@ -20,21 +20,17 @@ def __init__(self, settings: dict):
 
     @db_session
     def has_authorized_access(self, _, **kwargs) -> bool:
-        id_ = (request.json or request.form).get("id")
-        pass_ = (request.json or request.form).get("password")
-        if id_ is None or pass_ is None:
-            abort(401, "User not authorized")
+        token = request.headers.get("Authorization", request.cookies.get('token'))
 
-        user = User.get_s(id_)
+        if token is None:
+            abort(401)
 
-        if user is None:
-            abort(401, "Unknown user ID!")
+        parsed_token = jwt_service.verify_token(token, True)
 
-        if not user.email_verified:
-            abort(401, "Account needs verification")
```

Comment:
Should not be removed since we need to assert the account was owned by the email owner